### ISSUE TITLE
Tiny SFT loss

### ISSUE THREAD

**Issue by guoyd1999** [Gen-Verse/dLLM-RL#37]
Hi,

I use qwen3-0.6B-base as the base model and use the semi-ar training method to fine-tune it. The training data is demon_openr1math. training params are:

training:
    gradient_checkpointing_enable: True # if the sequence is very larger, set as True
    gradient_accumulation_steps: 2
    batch_size_lm: 24 # the total batch size is num_node * num_gpu_per_node * gradient_accumulation_steps * batch_size_lm
    mixed_precision: "bf16"
    enable_tf32: True
    seed: 10086
    num_train_epochs: 1
    max_grad_norm: 1
    method: "semi-ar" # "random_masking""semi-ar""ar" 3 different methods to choose
    lower_p: 0.1
    upper_p: 0.9
    block_size: 16 # use for semi-ar
    mask_times_per_sample: 35 # for random_masking
    post_num: 0 # number of pad token need to be trained for each data point
    max_gen_length: 1024
    max_prompt_len: 784

optimizer:
    name: adamw
    params: # default adamw params
        learning_rate: 5e-6
        scale_lr: False # scale learning rate by total batch size
        beta1: 0.9
        beta2: 0.999.

However, the training loss turns into very small after tens of training steps (e.g., 0.005). Can you tell me why?

**Comment by guoyd1999** [comment]
when I sft it on a larger dataset, its loss drops into ~5e-6, which is very weird: tensor(7.1824e-06, device='cuda:5', dtype=torch.bfloat16,                               
       grad_fn=<DivBackward0>)                                                          
tensor(0.0001, device='cuda:4', dtype=torch.bfloat16, grad_fn=<DivBackward0>)           
tensor(1.1444e-05, device='cuda:6', dtype=torch.bfloat16,                               
       grad_fn=<DivBackward0>)
tensor(2.6077e-06, device='cuda:7', dtype=torch.bfloat16,
       grad_fn=<DivBackward0>)
tensor(8.1658e-06, device='cuda:1', dtype=torch.bfloat16,
       grad_fn=<DivBackward0>)
tensor(5.9605e-06, device='cuda:0', dtype=torch.bfloat16,
       grad_fn=<DivBackward0>)
tensor(5.5730e-06, device='cuda:2', dtype=torch.bfloat16,
       grad_fn=<DivBackward0>)

**Comment by yinjjiew** [comment]
I think the loss drop very fast is reasonable right? You are changing AR model into block diffusion model. When you turn to larger dataset, did you change batch_size_lm or gradient_checkpointing_enable? Larger batch size here can make the loss appeared smaller. 
A good way to check if your training is correct, is to check the output of your trained model.
